flymake-ruff.el

```diff
@@ -20,19 +20,40 @@
   :group 'flymake-ruff
   :type 'string)
 
+(defcustom ruff-program-args
+  '("--format" "text" "--exit-zero" "--quiet" "-")
+  "Flags to be given to \"ruff\"."
+  :group 'flymake-ruff
+  :type '(list string))
+
 (defvar flymake-ruff--output-regex "\\(.*\\):\\([0-9]+\\):\\([0-9]+\\): \\([A-Z0-9]+\\) \\(.*\\)")
 
-;;TODO: check if exist a config file like pyproject.toml or ruff.toml and use it with --config=path arg
+(defconst flymake-ruff--default-configs
+  '("pyproject.toml" "ruff.toml" ".ruff.toml")
+  "Default configuration files supported by Ruff.")
+
 (defun flymake-ruff--check-buffer ()
   "Generate a list of diagnostics for the current buffer."
   (let ((code-buffer (current-buffer))
         (code-content (buffer-substring-no-properties (point-min) (point-max)))
         (dxs '()))
     (with-temp-buffer
       (insert code-content)
-      ;; call-process-region will run the program and replace current buffer
-      ;; with its stdout, that's why we need to run it in a temporary buffer
-      (call-process-region (point-min) (point-max) flymake-ruff-program t t nil "--format" "text" "--exit-zero" "--quiet" "-")
+      (let* ((config (car
+                      (seq-filter
+                       (lambda (f) (file-readable-p f))
+                       (mapcar
+                        (lambda (f)
+                          (file-name-concat
+                           (project-root (project-current)) f))
```

Comment:
```
You should depend on (emacs "28.1") if you need `project-root'.
```

Reply:
I changed it so that the package depends on project 0.3.0. You can install it from GNU ELPA, so if they can install `flymake-ruff`, they should also be able to install the latest version of `project`. Note that the latest version of `project.el` still supports Emacs 26.1.

Does that make sense?

Reply:
Yes, that's should works :)